### PR TITLE
[BACKLOG-40063] - Address review comments

### PR DESCRIPTION
**Pull request by soagarwal1**
@pentaho/hoth Please review
PR review fixes of https://github.com/pentaho/metastore/pull/89

### CODE REVIEW

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 1m 38s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox
```
---
:ok_hand: All tests passed!
 
Tests run: **121**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/metastore/job/PR-92/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by rmansoor — APPROVED** [review]

**Review by NJtwentyone — APPROVED** [review]

File: src/test/java/org/pentaho/metastore/test/MetaStoreFactoryTest.java [hunk 1]
```diff
@@ -560,15 +560,20 @@ private List<DimensionAttribute> generateAttributes() {
...
+    MetaStoreFactory<InheritedElement> factory =
+      new MetaStoreFactory<>( InheritedElement.class, metaStore, "custom" );
+    InheritedElement element = new InheritedElement( ANOTHER, ATTR );
+    element.setName( "name" );
+
+    assertEquals( 0, factory.getElements().size() );
+
+    factory.saveElement( element );
+
+    List<InheritedElement> elements = factory.getElements();
```
Comment (dcleao) [inline src/test/java/org/pentaho/metastore/test/MetaStoreFactoryTest.java:574]: It would be nice to also test that fields of the subclass are serialized as expected.

Also, another test is to make sure that duplicate/overridden private fields also work. Would better create another field in the base class for this test, perhaps.

**Review by dcleao — CHANGES_REQUESTED** [review]

File: src/test/java/org/pentaho/metastore/test/MetaStoreFactoryTest.java (hunk 1 above)
Comment (soagarwal1) [reply src/test/java/org/pentaho/metastore/test/MetaStoreFactoryTest.java:579]: Addressed

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 1m 32s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox
```
---
:ok_hand: All tests passed!
 
Tests run: **121**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/metastore/job/PR-92/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by dcleao — APPROVED** [review]
